### PR TITLE
add a func to merge multiple Strategic Merge Patches into one patch

### PR DESCRIPTION
**Pull request by mengqiy**
This PR is adding a function similar to https://github.com/evanphx/json-patch/blob/9fa11df836c7537e8ad3d1639fa0d396559a7bab/merge.go#L95-L100
It merges multiple SMPs and yield another SMP.

This PR has no risk to existing SMP code.

```release-note
NONE
```
/cc @apelisse
/assign @pwittrock

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go
Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go]: I'll rephrase. Does it make sense to merge 3 patches into 1 patch (still a patch)?

Comment (mengqiy) [reply staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go]: To merge 3 patches, you will need to do something like `merge2Patches(merge2Patches(patch1, patch2), patch3)`.
Merging 2 patches at once makes sense to me, but more than 2 patches at once does NOT.

Not sure if this answers your question.

Comment (mengqiy) [reply staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go]: > Does it make sense to merge 3 patches into 1 patch (still a patch)?

As long as we can make sure that `(base + P1) + P2 = base + (P1 + P2)`, this make sense to me.
Any other concerns?

Comment (apelisse) [reply staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go]: Nope

**Review by apelisse — APPROVED** [review]

**Comment by mengqiy** [comment]
/hold

Discussed offline a bit, it may be a little better if it takes more than 2 patches.

File: staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go
Comment (apelisse) [inline staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go]: Looks great.

**Review by apelisse — APPROVED** [review]
/lgtm

**Comment by mengqiy** [comment]
Sent https://github.com/kubernetes/kubernetes/pull/62148 to fix the failing test to unblock me...

**Pull request merged**
